modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheContext.java

```diff
@@ -2350,6 +2352,29 @@ public boolean hasContinuousQueryListeners(@Nullable IgniteInternalTx tx) {
             contQryMgr.notifyContinuousQueries(tx) && !F.isEmpty(contQryMgr.updateListeners(false, false));
     }
 
+    /**
+     * Apply changes from {@link SchemaAddQueryEntityOperation}
+     *
+     * @param op Add query entity schema operation.
+     */
+    public void onSchemaAddQueryEntity(SchemaAddQueryEntityOperation op) {
+        CacheConfiguration oldCfg = cacheCfg;
+
+        if (oldCfg != null) {
+            CacheConfiguration newCfg = new CacheConfiguration(oldCfg);
+
+            newCfg.setQueryEntities(Collections.singletonList(op.entity()));
```

Comment:
We have the same code inside `DynamicCacheDescriptor#schemaChangeFinish` and `GridCacheContextInfo#onSchemaAddQueryEntity` - let's deduplicate it and write in one place.

Reply:
I suggest to move this code to GridCacheUtils